apps/examples/testcase/le_tc/kernel/tc_task.c

```diff
@@ -414,6 +416,114 @@ static void tc_task_prctl(void)
 	TC_ASSERT_EQ_ERROR_CLEANUP("prctl", strncmp(getname, setname, CONFIG_TASK_NAME_SIZE), 0, get_errno(), prctl(PR_SET_NAME, oldname, 0, 0, 0));
 
 	prctl(PR_SET_NAME, oldname, 0, 0, 0);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+#ifdef CONFIG_MESSAGING_IPC
+	const char *portname = "testPort";
+	pid_t pid;
+	int recv_cnt = 0;
+	int *data_arr;
+	struct sched_param param;
+
+	/* saving receiver information */
+	pid = getpid();
+	ret_chk = sched_getparam(pid, &param);
+	TC_ASSERT_EQ("sched_getparam", ret_chk, OK);
+
+	ret_chk = prctl(PR_MSG_SAVE, portname, pid, param.sched_priority);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+	/* Reading receiver information */
+
+	ret_chk = prctl(PR_MSG_READ, getname, data_arr, recv_cnt);
+	TC_ASSERT_NEQ("prctl", data_arr, NULL);
+	TC_ASSERT_NEQ("prctl", recv_cnt, ERROR);
+
+	/* Saving receiver information */
+
+	ret_chk = prctl(PR_MSG_SAVE, portname, pid, param.sched_priority);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+	/* Removing receiver information */
+
+	ret_chk = prctl(PR_MSG_REMOVE, portname);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+#endif
+#ifdef CONFIG_TASK_MONITOR
+	ret_chk = prctl(PR_MONITOR_REGISTER, SEC_2);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+	ret_chk = prctl(PR_MONITOR_UPDATE, NULL);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+#endif
+#ifdef CONFIG_PREFERENCE
+	int test = 0;
+	bool *existing = false;
+	void *value = &test;
+	preference_data_t data;
+
+	data.key = TEST_STRING;
+	data.type = PRIVATE_PREFERENCE;
+	data.attr.type = PREFERENCE_TYPE_INT;
+	data.attr.len = sizeof(int);
+	data.value = value;
+
+	/* Set preference with prctl */
+
+	ret_chk = prctl(PR_SET_PREFERENCE, &data);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+	/* Get preference with prctl */
+
+	ret_chk = prctl(PR_GET_PREFERENCE, &data);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+	data.key = TEST_STRING;
+	data.type = PRIVATE_PREFERENCE;
+	data.attr.type = PREFERENCE_TYPE_BOOL;
+
+	/* Get preference with prctl */
+
+	ret_chk = prctl(PR_GET_PREFERENCE, &data);
+	TC_ASSERT_NEQ("prctl", data.type, ERROR);
+
+	/* Remove preference with prctl */
+
+	ret_chk = prctl(PR_REMOVE_PREFERENCE, PRIVATE_PREFERENCE, TEST_STRING);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+	/* Remove all preference with prctl */
+
+	ret_chk = prctl(PR_REMOVEALL_PREFERENCE, PRIVATE_PREFERENCE, NULL);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+#endif
+#ifdef CONFIG_SYSTEM_REBOOT_REASON
+	int reboot_reason_test_val = 100;
+
+	/* Clear, Write & Read the written reboot reason value */
+
+	ret_chk = prctl(PR_REBOOT_REASON_CLEAR);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+	ret_chk = prctl(PR_REBOOT_REASON_WRITE, reboot_reason_test_val);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+	ret_chk = prctl(PR_REBOOT_REASON_READ);
```

Comment:
Is this tc passed?
When read the reboot reason, it returns the reboot reason value, not OK.
So I think we should check 100 instead of OK.
Please check this.

Reply:
Like I said earlier, currently it is only supported by realtek board. So to avoid unnecessary ASSERTION during testcase check, I have added GEQ check.  It can be updated later on.
What is your opinion ?

Reply:
If a board will not support the reboot reason, then we should not enable `CONFIG_SYSTEM_REBOOT_REASON`.
In that case, this tc will not be tested. (It will be build error if a boart doesn't support the reboot reason, but enabling that config. Because in `prctl`, we cannot find the `up_reboot_reason_xxx()`s.)
So I think we don't need to consider

Reply:
As discussed offline, there is issue with reboot reason return value in the module itself. (as per reboot reason sample app on latest code)
I will update the handling of the test case return value to:
`TC_ASSERT_EQ("prctl", ret_chk, reboot_reason_test_val)`

Reply:
As discussed offline, reboot reason works well. Please modify the tc.
If you have any problem, please let me know.

Reply:
The reboot reason value written to the BKUP_REG only reflects after reboot. Otherwise it shows the previous reboot reason value. So I have mentioned in the comments the reason for a TC_ASSERT_GEQ check and did not modify it. This TC is only to cover code.